test/dynamo/test_functions.py

```diff
@@ -333,56 +331,37 @@ def test_methodcall3(a, b):
         return constant3(a, b=1.0) + b
 
     def test_is_integer(self):
-        @torch._dynamo.optimize(nopython=True)
-        def forward(t, m):
-            return t if m.is_integer() else 2 * t
-
-        t = torch.tensor([1])
-        assert forward(t, 1.0).item() == 1
-        assert forward(t, 1.5).item() == 2
-
-        assert forward(t, np.float32(1.0)).item() == 1
-        assert forward(t, np.float64(1.5)).item() == 2
-
-    @parametrize("method", ("as_integer_ratio", "bit_length", "conjugate"))
-    def test_int_methods(self, method):
-        @torch._dynamo.optimize(nopython=True)
-        def forward(t, m):
-            return t if getattr(m, method)() else 2 * t
-
-        t = torch.tensor([1])
-        forward(t, 1)
-
-    @parametrize("method", ("as_integer_ratio", "conjugate", "hex", "is_integer"))
-    def test_float_methods(self, method):
-        @torch._dynamo.optimize(nopython=True)
-        def forward(t, m):
-            return t if getattr(m, method)() else 2 * t
-
-        t = torch.tensor([1])
-        forward(t, 1.0)
-
-    def test_fraction_as_integer_ratio(self):
-        @torch._dynamo.optimize(nopython=True)
+        @torch.compile(backend="eager", fullgraph=True)
         def forward(t, m):
-            return t if m.as_integer_ratio()[1] else 2 * t
+            return 2 * t if m.is_integer() else t
 
         t = torch.tensor([1])
-        forward(t, fractions.Fraction(1))
+        self.assertEqual(forward(t, 1.0).item(), 2)
+        self.assertEqual(forward(t, 1.5).item(), 1)
 
-    @pytest.mark.xfail(
-        reason=(
-            "Fraction.conjugate() raises torch._dynamo.exc.Unsupported: "
-            "call_function UserDefinedClassVariable"
-        )
+    @parametrize(
+        "method, num_type",
+        (
+            ("as_integer_ratio", int),
+            ("bit_length", int),
+            ("conjugate", int),
+            ("as_integer_ratio", float),
+            ("conjugate", float),
+            ("hex", float),
+            ("is_integer", float),
+        ),
     )
-    def test_fraction_conjugate(self):
-        @torch._dynamo.optimize(nopython=True)
+    def test_number_method(self, method, num_type):
+        @torch.compile(backend="eager", fullgraph=True)
         def forward(t, m):
-            return t if m.conjugate() else 2 * t
-
-        t = torch.tensor([1])
-        forward(t, fractions.Fraction(1))
+            return 2 * t if getattr(m, method)() else t
+
+        for i in 1, 2.5:
```

Comment:
this, while correct, is so cursed. Let's be a bit more explicit.
```suggestion
        for i in (1, 2.5):
```

Reply:
Done, even though I'm skeptical it's really cursed. :-)